### DISCUSSION TITLE
Soften up python dependancies...?

### DISCUSSION THREAD

**Discussion by farchord** [darrenburns/posting#32] [General]
Good afternoon!

I'm looking into packaging Posting for Fedora, but I'm hitting a wall regarding the package requiring very specific package versions. For example:

click-default-group ( == 1.2.4)
pydantic (==2.7.3)
textual (==0.72)
textual[syntax] (==0.72)
xdg-base-dirs (==6.0.1)

Would it be possible to relax those requirements a bit? I guess I could alter the .toml to soften the requirements a bit but I wanted to see beforehand if you'd be willing to look at it.

Thank you!

**Comment by darrenburns** [comment]
Thanks for looking into packaging for Fedora.

To be honest I'm not sure the best approach to take here. When writing _libraries_, I always favour loose dependency constraints.

When users install Posting via `pipx`, the latest version of dependencies are used and this means a user which installs tomorrow could get different dependencies vs a user who installs today, even if no new Posting version is released.

I'd like everyone who installs Posting version X.Y.Z to get the exact same code.

I'm not sure how packaging on Fedora works at all to be honest, is there some shared environment which is causing these conflicts?

**Comment by alansill** [comment]
Although it's ugly, I have been able to get posting working by using conda to create and activate a Python 3.12 environment (`conda create -n py312 python=3.12` followed by `conda activate py312`, then download the latest released pipx.pyz file and use it as per the documentation (`python pipx.pyz ensurepath` and then either sign out and sign back in or execute the last line in your .bashrc file that was created by this command, and finally `python pipx.pyz install posting`).
